person_server.go

```diff
@@ -58,8 +72,91 @@ func (s *Server) addPerson(w http.ResponseWriter, r *http.Request) {
 		handleError(err, w, http.StatusUnprocessableEntity)
 		return
 	}
+	w.WriteHeader(http.StatusCreated)
+	json.NewEncoder(w).Encode(p)
+}
+
+func (s *Server) getPersons(w http.ResponseWriter, r *http.Request) {
+	if idStr := getQueryParam(r, "id"); idStr != "" {
```

Comment:
there should be two different endpoints:
/person/{id} - get by id
/person?name=123&phone=0737... - get by params. it should be able to find by any combination of passed query params(without params, only by name, only by phone, by both (logical operator - AND))